src/python/gudhi/off_utils.pyx

```diff
@@ -25,21 +26,91 @@ __license__ = "MIT"
+    :rtype: tuple(int, string)
+
+    Raises:
+        ValueError: If the file does not respect the OFF file format.
+    """
+    # Must start at -1 as first line (aka. 0), must be at least read
+    skiprows = -1
+    nb_vertices = -1
+    with open(off_file) as input_file:
+        skiprows, line = _get_next_line(skiprows, input_file)
+        # First line should be "OFF" (3d case) or "nOFF" (dD case)
+        if line.lower().startswith("off"):
+            dim = 3
+        elif line.lower().startswith("4off"):
+            dim = 4
+        elif line.lower().startswith("noff"):
+            # "nOFF" case, next line is the dimension
+            # can also contain nb_vertices, nb_faces nb_edges (can also be on the next line)
+            skiprows, line = _get_next_line(skiprows, input_file)
+            digits = [int(s) for s in line.split() if s.isdigit()]
+            dim = digits[0]
+            if len(digits) > 1:
+                nb_vertices = digits[1]
+                # nb_faces =  digits[2]
+                # nb_edges =  digits[3] # not used - can be ignored
+                # nb_cells =  digits[4]
+        else:
+            raise ValueError(f"Inconsistent OFF header for {off_file}, got '{line.rstrip()}', should be 'OFF', '4OFF' or 'nOFF'")
+        
+        # nb_vertices can be already set by "nOFF" case, when 'dim nb_vertices nb_faces nb_edges' on the same line
+        if nb_vertices < 0:
+            # Number of points is the first number ("OFF" case) or the second one ("nOFF" case) of the second line
+            skiprows, line = _get_next_line(skiprows, input_file)
+            digits = [int(s) for s in line.split() if s.isdigit()]
+            nb_vertices = digits[0]
+            # nb_faces =  digits[1]
+            # nb_edges =  digits[2] # not used - can be ignored
+            # nb_cells =  digits[3]
+        # Increment skiprows if comments between header and the first point
+        skiprows, line = _get_next_line(skiprows, input_file)
+        return skiprows, dim, nb_vertices
+
 def read_points_from_off_file(off_file=''):
     """Read points from an `OFF file <fileformats.html#off-file-format>`_.
 
     :param off_file: An OFF file style name.
     :type off_file: string
 
     :returns:  The point set.
-    :rtype: List[List[float]]
+    :rtype: numpy.ndarray
     """
-    if off_file:
-        if os.path.isfile(off_file):
-            return read_points_from_OFF_file(off_file.encode('utf-8'))
-        else:
-            raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT),
-                                    off_file)
+    skiprows, dim, nb_points = _read_off_file_header(off_file)
+    print(skiprows, dim, nb_points)
+    # usecols=list(range(dim)) stands here to avoid comments at the end of line
+    # or colors that can be added in RGB format after the points, the faces, ...
+    return np.loadtxt(off_file, dtype=np.floating, comments='#', skiprows=skiprows,
```

Comment:
loadtxt accepts a file instead of a filename, would it work to "reuse" the file we already opened?

Reply:
Done on [e819582](https://github.com/GUDHI/gudhi-devel/pull/1008/commits/e81958260af05ceac0b2f2c42e329c479be3a09b)